code/game/objects/effects/aliens.dm

```diff
@@ -59,31 +59,32 @@
 	. = ..()
 	if(ishuman(AM))
 		var/mob/living/carbon/human/H = AM
-		var/armor_block
-		if(TIMER_COOLDOWN_CHECK(H, COOLDOWN_ACID))
-			return
-		TIMER_COOLDOWN_START(H, COOLDOWN_ACID, 1 SECONDS)
-		if(!H.lying_angle)
-			to_chat(H, "<span class='danger'>Your feet scald and burn! Argh!</span>")
-			if(!(H.species.species_flags & NO_PAIN))
-				H.emote("pain")
-			H.next_move_slowdown += slow_amt
-			var/datum/limb/affecting = H.get_limb("l_foot")
-			armor_block = H.run_armor_check(affecting, "acid")
-			if(istype(affecting) && affecting.take_damage_limb(0, rand(14, 18), FALSE, FALSE, armor_block, TRUE))
-				UPDATEHEALTH(H)
-				H.UpdateDamageIcon()
-			affecting = H.get_limb("r_foot")
-			armor_block = H.run_armor_check(affecting, "acid")
-			if(istype(affecting) && affecting.take_damage_limb(0, rand(14, 18), FALSE, FALSE, armor_block, TRUE))
-				UPDATEHEALTH(H)
-				H.UpdateDamageIcon()
-		else
-			armor_block = H.run_armor_check("chest", "acid")
-			H.take_overall_damage(0, rand(12, 14), armor_block)
-			UPDATEHEALTH(H)
-			to_chat(H, "<span class='danger'>You are scalded by the burning acid!</span>")
+		H.acid_spray_crossed(slow_amt)
+
+/mob/living/carbon/human/proc/acid_spray_crossed(slow_amt)
+	if(TIMER_COOLDOWN_CHECK(src, COOLDOWN_ACID))
+		return
 
+	TIMER_COOLDOWN_START(src, COOLDOWN_ACID, 1 SECONDS)
+	if(HAS_TRAIT(src, TRAIT_FLOORED))
+		take_overall_damage(0, rand(12, 14), run_armor_check("chest", "acid"))
+		UPDATEHEALTH(src)
+		to_chat(src, "<span class='danger'>You are scalded by the burning acid!</span>")
+		return
+	to_chat(src, "<span class='danger'>Your feet scald and burn! Argh!</span>")
+	if(!(species.species_flags & NO_PAIN))
+		emote("pain")
+	next_move_slowdown += slow_amt
+	var/datum/limb/affecting = get_limb("l_foot")
```

Comment:
for these `get_limb()` please use the associated defines. `BODY_ZONE_PRECISE_L_FOOT` in this example